pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -405,6 +415,21 @@ func (c *Controller) addNodeRoute(nodeName string, node *corev1.Node) error {
 		// Does not help to return an error and trigger controller retries.
 		return nil
 	}
+
+	nodesHaveSamePodCIDR, _ := c.installedNodes.ByIndex(nodeRouteInfoPodCIDRIndexName, node.Spec.PodCIDR)
+	// PodCIDRs can be released from deleted Nodes and allocated to new Nodes. For server side, it won't happen that a
+	// PodCIDR is allocated to more than one Node at any point. However, for client side, if a resync happens to occur
```

Comment:
more than one Node -> more than one Nodes
You might want to change the commit message too.

Reply:
"more than one Node" is correct

Reply:
Seems you are right!